client/app/components/weather/weather.controller.js

```diff
@@ -0,0 +1,22 @@
+class WeatherController {
+  constructor($scope) {
+    'ngInject';
+
+    var vm = this;
+    vm.name = 'weather';    
+    vm.zipCode = '';
+
+    $scope.$watch(function watchZipCode(){
```

Comment:
You don't need `$scope` at all. Instead of this just use setters. And you also shouldn't use `$broadcast` to implement pub/sub. In your case you should just call service method:

```
class WeatherController{
    constructor(forecasts) {
       this.forecasts = forecasts;
    }
    get zipCode() {
        return this._zipCode;
    }
    set zipCode(val) {
        this._zipCode = val;
        if (val.length !== 5) {
            this.forecast = [];
        }

        this.forecasts.refresh();
    }
}
```